awssdkgo module panics when a non-supported service is used

**Describe the bug**
When a wrapped AWS session is used with an AWS service not supported by APM, (today only dynamo, sqs, sns, and s3 are supported), it panics in the `complete()` function at this line, if more than 1 AWS call is executed in that session:
https://github.com/elastic/apm-agent-go/blob/master/module/apmawssdkgo/session.go#L199

I think the cause is because the `complete()` function does not have the same check as the other two functions for unsupported services:
```go
        spanSubtype := req.ClientInfo.ServiceName
	spanType, ok := serviceTypeMap[spanSubtype]
	if !ok {
		return
	}
```
Since it is missing this check, these lines end up fetching a different span from the context (in my case, one I create manually) and close it: https://github.com/elastic/apm-agent-go/blob/master/module/apmawssdkgo/session.go#L193-L197

On the second call, this function then fetches the same span from the context, but since it has already been closed, the `span.SpanData` (where span.Context is) is set to nil, so the `span.Context.SetHTTPStatus` panics.

**To Reproduce**
```go
package main

import (
	"context"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/ses"
	"go.elastic.co/apm"
	"go.elastic.co/apm/module/apmawssdkgo"
)

func main() {
	charset := "UTF-8"
	sess := apmawssdkgo.WrapSession(session.Must(session.NewSession(&aws.Config{Region: aws.String("us-east-1")})))


	// Create an SES session.
	svc := ses.New(sess)

	// Assemble the email.
	input := &ses.SendEmailInput{
		Source: aws.String("test@arthur.ai"),
		Destination: &ses.Destination{
			CcAddresses: []*string{},
			ToAddresses: []*string{aws.String("ian@arthur.ai")},
		},
		Message: &ses.Message{
			Body: &ses.Body{
				Text: &ses.Content{
					Charset: aws.String(charset),
					Data:    aws.String("test panic"),
				},
			},
			Subject: &ses.Content{
				Charset: aws.String(charset),
				Data:    aws.String("test panic"),
			},
		},
	}

	tx := apm.DefaultTracer.StartTransaction("send-email", "test-tx")
	span := tx.StartSpan("test-span", "send-email", nil)
	defer span.End()
	ctx := apm.ContextWithSpan(context.Background(), span)

	// this call ends the span
	_, _ = svc.SendEmailWithContext(ctx, input)

	// this call panics
	_, _ = svc.SendEmailWithContext(ctx, input)
}
```

Produces the following panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1a8 pc=0x53a8966]

goroutine 151 [running]:
go.elastic.co/apm/module/apmawssdkgo.complete(0xc000b55400)
        /Users/ianmcgraw/go/pkg/mod/go.elastic.co/apm/module/apmawssdkgo@v1.12.0/session.go:199 +0xc6
github.com/aws/aws-sdk-go/aws/request.(*HandlerList).Run(0xc000b55728, 0xc000b55400)
        /Users/ianmcgraw/go/pkg/mod/github.com/aws/aws-sdk-go@v1.38.14/aws/request/handlers.go:267 +0x98
github.com/aws/aws-sdk-go/aws/request.(*Request).Send.func1(0xc000b55400)
        /Users/ianmcgraw/go/pkg/mod/github.com/aws/aws-sdk-go@v1.38.14/aws/request/request.go:515 +0x39
github.com/aws/aws-sdk-go/aws/request.(*Request).Send(0xc000b55400, 0x0, 0x0)
        /Users/ianmcgraw/go/pkg/mod/github.com/aws/aws-sdk-go@v1.38.14/aws/request/request.go:532 +0x230
github.com/aws/aws-sdk-go/service/ses.(*SES).SendEmailWithContext(0xc0006929c0, 0x61644a0, 0xc000a71bc0, 0xc000a42070, 0x0, 0x0, 0x0, 0xc000d92801, 0x101000006113520, 0x0)
        /Users/ianmcgraw/go/pkg/mod/github.com/aws/aws-sdk-go@v1.38.14/service/ses/api.go:4463 +0xae
arthur-api-service/pkg/api/service.sendSesEmail(0x61644a0, 0xc000a71bc0, 0xc000a22200, 0x12, 0x5cc90de, 0x1a, 0x5cb7f63, 0x11, 0xc000a60dc0, 0x2bd, ...)
        /Users/ianmcgraw/dev/arthur-api-service/pkg/api/service/notification_service.go:167 +0x528
arthur-api-service/pkg/api/service.SendEmail(0x61644a0, 0xc000a71bc0, 0xc000a22200, 0x12, 0x5cc90de, 0x1a, 0x5cb7f63, 0x11, 0xc000a60dc0, 0x2bd, ...)
        /Users/ianmcgraw/dev/arthur-api-service/pkg/api/service/notification_service.go:127 +0x1b2
arthur-api-service/pkg/api/service.SendUserInviteEmail(0x61644a0, 0xc000a71bc0, 0x5645281d38e70813, 0xc517cdbcfef6ddb8, 0x0, 0x0, 0x0, 0xf34918d0702ed1bf, 0xe65fdaf8c14c67aa, 0x5340c4a1598589fc, ...)
        /Users/ianmcgraw/dev/arthur-api-service/pkg/api/service/users_service.go:146 +0x471
arthur-api-service/pkg/api/service.SendEmailInvites.func1(0xc0001f71a0, 0x61644a0, 0xc000a71bc0, 0xc0001f6b40, 0x0, 0x0, 0xc000c9f500)
        /Users/ianmcgraw/dev/arthur-api-service/pkg/api/service/users_service.go:105 +0x18a
created by arthur-api-service/pkg/api/service.SendEmailInvites
        /Users/ianmcgraw/dev/arthur-api-service/pkg/api/service/users_service.go:97 +0xcd

Process finished with the exit code 2
```

thanks APM team

--- a/module/apmawssdkgo/session.go
+++ b/module/apmawssdkgo/session.go
@@ -75,12 +75,11 @@ type service interface {
 }
 
 func build(req *request.Request) {
-	spanSubtype := req.ClientInfo.ServiceName
-	spanType, ok := serviceTypeMap[spanSubtype]
-	if !ok {
+	if !supportedRequest(req) {
 		return
 	}
 
+	spanSubtype := req.ClientInfo.ServiceName
 	if spanSubtype == serviceSNS && !supportedSNSMethod(req) {
 		return
 	}
@@ -96,6 +95,7 @@ func build(req *request.Request) {
 
 	// The span name is added in the `send()` function, after other
 	// handlers have generated the necessary information on the request.
+	spanType := serviceTypeMap[spanSubtype]
 	span := tx.StartSpan("", spanType, apm.SpanFromContext(ctx))
 	if !span.Dropped() {
 		ctx = apm.ContextWithSpan(ctx, span)
@@ -121,9 +121,7 @@ func send(req *request.Request) {
 		return
 	}
 
-	spanSubtype := req.ClientInfo.ServiceName
-	_, ok := serviceTypeMap[spanSubtype]
-	if !ok {
+	if !supportedRequest(req) {
 		return
 	}
 
@@ -137,6 +135,7 @@ func send(req *request.Request) {
 		svc service
 		err error
 	)
+	spanSubtype := req.ClientInfo.ServiceName
 	switch spanSubtype {
 	case serviceS3:
 		svc = newS3(req)
@@ -189,9 +188,7 @@ func send(req *request.Request) {
 }
 
 func complete(req *request.Request) {
-	spanSubtype := req.ClientInfo.ServiceName
-	_, ok := serviceTypeMap[spanSubtype]
-	if !ok {
+	if !supportedRequest(req) {
 		return
 	}
 
@@ -207,4 +204,9 @@ func complete(req *request.Request) {
 	if err := req.Error; err != nil {
 		apm.CaptureError(ctx, err).Send()
 	}
+}
+
+func supportedRequest(req *request.Request) bool {
+	_, ok := serviceTypeMap[req.ClientInfo.ServiceName]
+	return ok
 }

--- a/module/apmawssdkgo/session.go
+++ b/module/apmawssdkgo/session.go
@@ -189,6 +189,12 @@ func send(req *request.Request) {
 }
 
 func complete(req *request.Request) {
+	spanSubtype := req.ClientInfo.ServiceName
+	_, ok := serviceTypeMap[spanSubtype]
+	if !ok {
+		return
+	}
+
 	ctx := req.Context()
 	span := apm.SpanFromContext(ctx)
 	if span.Dropped() {

--- a/module/apmawssdkgo/session_test.go
+++ b/module/apmawssdkgo/session_test.go
@@ -23,15 +23,18 @@ import (
 	"bytes"
 	"context"
 	"testing"
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/athena"
 	"github.com/aws/aws-sdk-go/service/dynamodb"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"go.elastic.co/apm"
 	"go.elastic.co/apm/apmtest"
 )
 
@@ -170,4 +173,32 @@ func TestDynamoDB(t *testing.T) {
 		assert.Equal(t, tx.ID, span.ParentID)
 	}
 
+}
+
+func TestUnsupportedServices(t *testing.T) {
+	region := "us-west-2"
+	cfg := aws.NewConfig().
+		WithRegion(region).
+		WithDisableSSL(true).
+		WithCredentials(credentials.AnonymousCredentials)
+
+	session := session.Must(session.NewSession(cfg))
+	wrapped := WrapSession(session)
+	svc := athena.New(wrapped)
+
+	tx := apm.DefaultTracer.StartTransaction("send-email", "test-tx")
+	span := tx.StartSpan("test-span", "send-email", nil)
+	defer span.End()
+
+	ctx := apm.ContextWithSpan(context.Background(), span)
+	namedQuery := &athena.BatchGetNamedQueryInput{
+		NamedQueryIds: []*string{aws.String("query")},
+	}
+	// Setting a timeout so the request fails fast. This doesn't affect
+	// testing that we noop on unsupported services.
+	ctx, cancel := context.WithTimeout(ctx, time.Millisecond)
+	defer cancel()
+
+	svc.BatchGetNamedQueryWithContext(ctx, namedQuery)
+	assert.NotPanics(t, func() { svc.BatchGetNamedQueryWithContext(ctx, namedQuery) })
 }

